### PR TITLE
Add fork project support

### PR DESCRIPTION
**Pull request by fgouteroux**
- Forks a project into the user namespace of the authenticated user.
- Allows modification of the forked relationship between existing projects (for admins)

### CODE REVIEW

File: gitlab/__init__.py [hunk 1]
```diff
@@ -614,6 +614,13 @@ def Team(self, id=None, **kwargs):
         """
         return Team._get_list_or_object(self, id, **kwargs)
 
+    def ForkProject(self, id=None, **kwargs):
```
Comment (gpocentek) [inline gitlab/__init__.py]: The class name should be ProjectFork.

File: gitlab/__init__.py [hunk 2]
```diff
@@ -1376,3 +1404,12 @@ def Project(self, id=None, **kwargs):
...
                                                **kwargs)
+
+class ForkProject(GitlabObject):
+    _url = '/projects/fork/%(project_id)s'
+    canUpdate = False
+    canDelete = False
+    canList = False
+    canGet = False
+    requiredUrlAttrs = ['project_id']
+    requiredCreateAttrs = ['project_id']
```
Comment (gpocentek) [inline gitlab/__init__.py]: requiredCreateAttrs should not be defined. If it exists the CLI will not work (the project-id argument would be defined twice).

File: gitlab/__init__.py [hunk 3]
```diff
@@ -1376,3 +1404,12 @@ def Project(self, id=None, **kwargs):
         return TeamProject._get_list_or_object(self.gitlab, id,
                                                team_id=self.id,
                                                **kwargs)
+
+class ForkProject(GitlabObject):
```
Comment (gpocentek) [inline gitlab/__init__.py]: The function name should be ProjectFork.

**Comment by gpocentek** [comment]
Looks good but could you address the comments? Thanks.

**Comment by fgouteroux** [comment]
Thanks for your comments, I have updated the code. I also have re-synchronized the branch due to the updates between the two commits.

**Comment by gpocentek** [comment]
I've changed a lot of things last week and it looks like you merge didn't take that into account. could you have a look? Sorry for making your work harder...

**Comment by fgouteroux** [comment]
It will be more simple to create a new PR with the last updates.

**Pull request closed without merging**